### DISCUSSION TITLE
Accessing container form from a field

### DISCUSSION THREAD

**Discussion by robballantyne** [wintercms/winter#286] [Q&A] [answered]
It's probably simple but driving me mad. 

I have a nested form with a partial inside, from the partial i naturally have $field available, but I would like to figure out how to identify (and get data from) another field in that form. Is there a simple way to do this?

**Comment by bennothommo** [comment, marked as answer]
@robballantyne You could just query the model directly for the data if you know the field name.

For example, if the field is for the property `name`, you could just use `$model->name` in your partial.

**Reply by robballantyne** [reply]
Thanks Ben, i have had a bit of time to reflect and it turns out that I should get more sleep.

I don't need to do what I thought I did at all, but I did wonder why a field has no direct relationship with its form (unlike widgets). I won't spend any more time on this though. Thanks again!

**Reply by LukeTowers** [reply]
@robballantyne could you share what your initial use case was that led you to believe this was the solution and what you ultimately ended up doing instead? 

I do somewhat agree that it's strange that the field object doesn't have access to it's containing form, but it might not have helped you in this case either as the Form instance itself doesn't have access to the widget that spawned it (the NestedForm widget in this case)

**Reply by robballantyne** [reply]
Sure,

Re attached screenshot, I wanted to get the data from the richeditor (or any other field/widget) in the 'main' tab into the 'preview' tab without having to save the model. 

I now think I can just do this by refreshing the form and the data will be available in $model anyway, even if I haven't saved - If thats correct, I just need to apply some logic to do a form refresh when the preview tab is activated.

![wn-fields](https://user-images.githubusercontent.com/8018501/130942486-4dfa3ca1-0d62-48e0-80a2-6ce7c70a36c9.png)

Ps. excuse the ui, im developing for both platforms.

**Reply by LukeTowers** [reply]
@robballantyne you could try using the dependsOn option to trigger an update of your partial field based on the richeditor field.

**Reply by robballantyne** [reply]
@LukeTowers dependsOn is the right way, thanks. Simple things escape me at times. 

I've gone for a manual refresh, though using a checkbox as my richeditor wasnt causing a refresh. Also I probably don't want to re-load the whole view on every keystroke while it's out of view.

Happy with the results - Thanks again!

**Reply by LukeTowers** [reply]
@robballantyne your richeditor looks like a custom integration, in which case you need to make sure you're triggering the `change` event on the element that holds your data to trigger the dependsOn and trigger functionality

**Reply by LukeTowers** [reply]
Since you'd have to do that custom anyways, there's no reason why you couldn't debounce the event and only trigger it after x seconds of no typing.
